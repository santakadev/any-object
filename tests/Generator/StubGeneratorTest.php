<?php

namespace Santakadev\AnyObject\Tests\Generator;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Santakadev\AnyObject\Generator\StubGenerator;
use Santakadev\AnyObject\Tests\Generator\Generated\Any;
use Santakadev\AnyObject\Tests\TestData\ScalarTypes\StringObject;

class StubGeneratorTest extends TestCase
{
    public function test_generator(): void
    {
        $generator = new StubGenerator();
        $text = $generator->generate(StringObject::class);
        Approvals::verifyString($text);
        $test = Any::of(StringObject::class)->with(value: "test");
        $this->assertEquals("test", $test->value);
    }
}
