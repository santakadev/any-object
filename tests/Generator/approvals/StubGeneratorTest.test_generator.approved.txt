<?php

namespace Santakadev\AnyObject\Generator\Tests\Generated;

use Faker\Factory;
use Santakadev\AnyObject\Tests\TestData\ScalarTypes\StringObject;

final class StringObjectFactory
{
    public function with(string|ValueNotProvided $value = new ValueNotProvided): StringObject
    {
        if ($value instanceof ValueNotProvided) {
            $value = (Factory::create())->text();
        }

        return new StringObject($value);
    }

    public function build(): StringObject
    {
        return self::with();
    }
}
